test/Microsoft.AspNetCore.ResponseCompression.Tests/ResponseCompressionMiddlewareTest.cs

```diff
@@ -42,15 +63,39 @@ public void Options_HttpsDisabledByDefault()
         [Fact]
         public async Task Request_AcceptGzipDeflate_CompressedGzip()
         {
-            var response = await InvokeMiddleware(100, requestAcceptEncodings: new string[] { "gzip", "deflate" }, responseType: TextPlain);
+            var response = await InvokeMiddleware(100, requestAcceptEncodings: new[] { "gzip", "deflate" }, responseType: TextPlain);
+
+            CheckResponseCompressed(response, expectedBodyLength: 24, expectedEncoding: "gzip");
+        }
+
+        [Fact]
+        public async Task Request_AcceptBrotli_CompressedBrotli()
+        {
+            var response = await InvokeMiddleware(100, requestAcceptEncodings: new[] { "br" }, responseType: TextPlain);
+
+#if NET461
+            CheckResponseNotCompressed(response, expectedBodyLength: 100, sendVaryHeader: true);
+#elif NETCOREAPP2_2
+            CheckResponseCompressed(response, expectedBodyLength: 20, expectedEncoding: "br");
+#else
+#error Target frameworks need to be updated.
+#endif
+        }
+
+        [Theory]
+        [InlineData("gzip", "br")]
+        [InlineData("br", "gzip")]
+        public async Task Request_AcceptMixed_CompressedGzip(string encoding1, string encoding2)
```

Comment:
I don't see the opposite test where you configure the providers in the other order and ensure br is selected. (NETCOREAPP2_2 only)

Reply:
Yeah, that's still missing. I'm reluctant to copy-pasting the setup code more, but I'm not sure if there there's a good way to reuse the setup as there are quite a few permutations.